particle-iot/particle-cli
# feature/library-json

## Description

Adds support for a `--json` flag to return machine readable JSON from the following commands:

```
particle library search
particle library list
particle library view
```

## How to Test

verify help content:

```
particle library search --help
particle library list --help
particle library view --help
```

verify JSON output - run these commands while signed-in and while signed-out:

```
particle library search dotstar --json
particle library search DOESNOTEXIST --json
particle library list --json
particle library list --json --page 2
particle library list --json --page 2 --limit 50
particle library view dotstar --json
particle library view DOESNOTEXIST --json
particle library view dotstar --json --readme
particle library view DOESNOTEXIST --json --readme
particle library view dotstar --json --source
particle library view DOESNOTEXIST --json --source
particle library view dotstar --json --header
particle library view DOESNOTEXIST --json --header
```

in all cases you should see properly formatted JSON output via `stdout` with appropriate exit codes.

## Completeness
